### ISSUE TITLE
Terrain_TQO_Bug: Editor crashes on adding an Axis Aligned Box Shape component to a Terrain Spawner setup

### ISSUE THREAD

**Issue by jckand-amzn** [o3de/o3de#8934]
**Describe the bug**
Editor crashes on adding an Axis Aligned Box Shape component to a Terrain Spawner setup.

**Prerequisites**
1. Terrain Gem is enabled for the current project.

**Steps to reproduce**
1. Open a new or existing level in the Editor.
3. Click on the root Level entity, and add the Terrain World, Terrain World Renderer, and Terrain Debugging level components.
4. Create a new entity.
5. Add a Terrain Layer Spawner component and an Axis Aligned Box Shape component to the entity.

**Expected behavior**
Components are added free of errors.

**Actual behavior**
Editor crashes.

```
>	Terrain.Editor.dll!AZ::Vector3::GetX() Line 129	C++
 	Terrain.Editor.dll!SurfaceData::AabbContains2D(const AZ::Aabb & box, const AZ::Vector3 & point) Line 143	C++
 	Terrain.Editor.dll!Terrain::TerrainSystem::FindBestAreaEntityAtPosition(const AZ::Vector3 & position, AZ::Aabb & bounds) Line 893	C++
 	Terrain.Editor.dll!Terrain::TerrainSystem::MakeBulkQueries(const AZStd::span<AZ::Vector3 const ,-1> inPositions, AZStd::span<AZ::Vector3,-1> outPositions, AZStd::span<bool,-1> outTerrainExists, AZStd::span<AZStd::fixed_vector<AzFramework::SurfaceData::SurfaceTagWeight,16>,-1> outSurfaceWeights, AZStd::function<void __cdecl(AZStd::span<AZ::Vector3 const ,-1>,AZStd::span<AZ::Vector3,-1>,AZStd::span<bool,-1>,AZStd::span<AZStd::fixed_vector<AzFramework::SurfaceData::SurfaceTagWeight,16>,-1>,AZ::EntityId)> queryCallback) Line 333	C++
 	Terrain.Editor.dll!Terrain::TerrainSystem::GetHeightsSynchronous(const AZStd::span<AZ::Vector3 const ,-1> & inPositions, AzFramework::Terrain::TerrainDataRequests::Sampler sampler, AZStd::span<float,-1> heights, AZStd::span<bool,-1> terrainExists) Line 413	C++
 	Terrain.Editor.dll!Terrain::TerrainSystem::ProcessHeightsFromRegion(const AZ::Aabb & inRegion, const AZ::Vector2 & stepSize, AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)> perPositionCallback, AzFramework::Terrain::TerrainDataRequests::Sampler sampler) Line 1228	C++
 	Terrain.Editor.dll!AZStd::Internal::INVOKE<void (__cdecl AzFramework::Terrain::TerrainDataRequests::*)(AZ::Aabb const &,AZ::Vector2 const &,AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>,enum AzFramework::Terrain::TerrainDataRequests::Sampler)const ,AzFramework::Terrain::TerrainDataRequests * &,AZ::Aabb &,AZ::Vector2 const &,`Terrain::TerrainFeatureProcessor::UpdateHeightmapImage'::`16'::<lambda_1> &,enum AzFramework::Terrain::TerrainDataRequests::Sampler &,void>(void(const AzFramework::Terrain::TerrainDataRequests::*)(const AZ::Aabb &, const AZ::Vector2 &, AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>, AzFramework::Terrain::TerrainDataRequests::Sampler) && f, AzFramework::Terrain::TerrainDataRequests * & arg0, AZ::Aabb & <args_0>, const AZ::Vector2 & <args_1>, Terrain::TerrainFeatureProcessor::UpdateHeightmapImage::__l16::<lambda_1> & <args_2>, AzFramework::Terrain::TerrainDataRequests::Sampler & <args_3>) Line 178	C++
 	Terrain.Editor.dll!AZStd::invoke<void (__cdecl AzFramework::Terrain::TerrainDataRequests::*)(AZ::Aabb const &,AZ::Vector2 const &,AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>,enum AzFramework::Terrain::TerrainDataRequests::Sampler)const ,AzFramework::Terrain::TerrainDataRequests * &,AZ::Aabb &,AZ::Vector2 const &,`Terrain::TerrainFeatureProcessor::UpdateHeightmapImage'::`16'::<lambda_1> &,enum AzFramework::Terrain::TerrainDataRequests::Sampler &>(void(const AzFramework::Terrain::TerrainDataRequests::*)(const AZ::Aabb &, const AZ::Vector2 &, AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>, AzFramework::Terrain::TerrainDataRequests::Sampler) && f, AzFramework::Terrain::TerrainDataRequests * & <args_0>, AZ::Aabb & <args_1>, const AZ::Vector2 & <args_2>, Terrain::TerrainFeatureProcessor::UpdateHeightmapImage::__l16::<lambda_1> & <args_3>, AzFramework::Terrain::TerrainDataRequests::Sampler & <args_4>) Line 56	C++
 	Terrain.Editor.dll!AZ::EBusEventProcessingPolicy::Call<void (__cdecl AzFramework::Terrain::TerrainDataRequests::*)(AZ::Aabb const &,AZ::Vector2 const &,AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>,enum AzFramework::Terrain::TerrainDataRequests::Sampler)const ,AzFramework::Terrain::TerrainDataRequests * &,AZ::Aabb &,AZ::Vector2 const &,`Terrain::TerrainFeatureProcessor::UpdateHeightmapImage'::`16'::<lambda_1> &,enum AzFramework::Terrain::TerrainDataRequests::Sampler &>(void(const AzFramework::Terrain::TerrainDataRequests::*)(const AZ::Aabb &, const AZ::Vector2 &, AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>, AzFramework::Terrain::TerrainDataRequests::Sampler) && func, AzFramework::Terrain::TerrainDataRequests * & iface, AZ::Aabb & <args_0>, const AZ::Vector2 & <args_1>, Terrain::TerrainFeatureProcessor::UpdateHeightmapImage::__l16::<lambda_1> & <args_2>, AzFramework::Terrain::TerrainDataRequests::Sampler & <args_3>) Line 438	C++
 	Terrain.Editor.dll!AZ::Internal::EBusContainer<AzFramework::Terrain::TerrainDataRequests,AzFramework::Terrain::TerrainDataRequests,0,0>::Dispatcher<AZ::EBus<AzFramework::Terrain::TerrainDataRequests,AzFramework::Terrain::TerrainDataRequests>>::Broadcast<void (__cdecl AzFramework::Terrain::TerrainDataRequests::*)(AZ::Aabb const &,AZ::Vector2 const &,AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>,enum AzFramework::Terrain::TerrainDataRequests::Sampler)const ,AZ::Aabb &,AZ::Vector2 const &,`Terrain::TerrainFeatureProcessor::UpdateHeightmapImage'::`16'::<lambda_1> &,enum AzFramework::Terrain::TerrainDataRequests::Sampler &>(void(const AzFramework::Terrain::TerrainDataRequests::*)(const AZ::Aabb &, const AZ::Vector2 &, AZStd::function<void __cdecl(unsigned __int64,unsigned __int64,AzFramework::SurfaceData::SurfacePoint const &,bool)>, AzFramework::Terrain::TerrainDataRequests::Sampler) && func, AZ::Aabb & <args_0>, const AZ::Vector2 & <args_1>, Terrain::TerrainFeatureProcessor::UpdateHeightmapImage::__l16::<lambda_1> & <args_2>, AzFramework::Terrain::TerrainDataRequests::Sampler & <args_3>) Line 1541	C++
 	Terrain.Editor.dll!Terrain::TerrainFeatureProcessor::UpdateHeightmapImage() Line 292	C++
 	Terrain.Editor.dll!Terrain::TerrainFeatureProcessor::ProcessSurfaces(const AZ::RPI::FeatureProcessor::RenderPacket & process) Line 417	C++
 	Terrain.Editor.dll!Terrain::TerrainFeatureProcessor::Render(const AZ::RPI::FeatureProcessor::RenderPacket & packet) Line 130	C++
 	Atom_RPI.Editor.dll!`AZ::RPI::Scene::CollectDrawPacketsJobs'::`6'::<lambda_1>::operator()() Line 608	C++
 	Atom_RPI.Editor.dll!AZ::JobFunction<`AZ::RPI::Scene::CollectDrawPacketsJobs'::`6'::<lambda_1>>::ArgHelper<`AZ::RPI::Scene::CollectDrawPacketsJobs'::`6'::<lambda_1>,0>::Process(AZ::RPI::Scene::CollectDrawPacketsJobs::__l6::<lambda_1> & function, AZ::Job & __formal) Line 71	C++
 	Atom_RPI.Editor.dll!AZ::JobFunction<`AZ::RPI::Scene::CollectDrawPacketsJobs'::`6'::<lambda_1>>::Process() Line 43	C++
 	EditorLib.dll!AZ::Internal::JobManagerBase::Process(AZ::Job * job) Line 31	C++
 	EditorLib.dll!AZ::Internal::JobManagerWorkStealing::ProcessJobsInternal(AZ::Internal::JobManagerWorkStealing::ThreadInfo * info, AZ::Job * suspendedJob, std::atomic<bool> * notifyFlag) Line 421	C++
 	EditorLib.dll!AZ::Internal::JobManagerWorkStealing::ProcessJobsWorker(AZ::Internal::JobManagerWorkStealing::ThreadInfo * info) Line 312	C++
 	EditorLib.dll!`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>::operator()() Line 655	C++
 	EditorLib.dll!AZStd::Internal::INVOKE<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>>(AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads::__l4::<lambda_1> && f) Line 209	C++
 	EditorLib.dll!AZStd::invoke<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>>(AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads::__l4::<lambda_1> && f) Line 56	C++
 	EditorLib.dll!AZStd::Internal::apply_impl<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>,std::tuple<>>(AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads::__l4::<lambda_1> && f, std::tuple<> && tupleObj, std::integer_sequence<unsigned __int64> __formal) Line 283	C++
 	EditorLib.dll!AZStd::apply<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>,std::tuple<>>(AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads::__l4::<lambda_1> && f, std::tuple<> && tupleObj) Line 291	C++
 	EditorLib.dll!`AZStd::thread::thread<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>>'::`2'::<lambda_1>::operator()() Line 46	C++
 	EditorLib.dll!AZStd::Internal::thread_info_impl<`AZStd::thread::thread<`AZ::Internal::JobManagerWorkStealing::CreateWorkerThreads'::`4'::<lambda_1>>'::`2'::<lambda_1>>::execute() Line 193	C++
 	EditorLib.dll!AZStd::Internal::thread_run_function(void * param) Line 38	C++
 	[External Code]	
```

**Found in Branch**
stabilization2205 installer

**Comment by jckand-amzn** [comment]
This does not occur if the Axis Aligned Bounding Box component is added to the entity first.

**Comment by mbalfour-amzn** [comment]
Fixed in stabilization.

**Comment by jckand-amzn** [comment]
Verified fix on stabilization/2205@af75b10590969bf7038bf4acb4ea10e1663bd420.
